src/sentry/uptime/subscriptions/subscriptions.py

```diff
@@ -351,8 +392,11 @@ def enable_project_uptime_subscription(
         disable_project_uptime_subscription(uptime_monitor)
         raise UptimeMonitorNoSeatAvailable(None)
 
-    uptime_monitor.update(status=ObjectStatus.ACTIVE)
     uptime_subscription = uptime_monitor.uptime_subscription
+    uptime_monitor.update(status=ObjectStatus.ACTIVE)
+
+    if detector := get_detector(uptime_monitor.uptime_subscription):
+        detector.update(enabled=True)
```

Comment:
Transaction here

Reply:
Should we have one here, or should we just put the calls to this function into the txn in the callstack above (in create / update)